manual/manual/tutorials/coreexamples.etex

```diff
@@ -289,6 +289,91 @@ let rec insert x btree =
                 else Node(y, left, insert x right);;
 \end{caml_example}
 
+
+\subsection{Record and variant disambiguation}
+( This subsection can be skipped on the first reading )
+
+Astute readers may have wondered what happens when two or more record
+fields or constructors share the same name
+
+\begin{caml_example*}{toplevel}
+type first  = { x:int; y:int; z:int }
+type middle = { x:int; z:int }
+type last   = { x:int };;
+type first_variant = A | B | C
+type last_variant  = A;;
+\end{caml_example*}
+
+The answer is that when confronted with multiple options, OCaml tries to
+use locally available information to disambiguate between the various fields
+and constructors. First, if the type of the record or variant is known,
+OCaml can pick unambiguously the corresponding field or constructor.
+For instance:
+
+\begin{caml_example}{toplevel}
+let look_at_x_then_y_then_z (r:first) =
```

Comment:
The function name doesn't match the implementation anymore?

Reply:
You are right, fixed.